src/context/featureFlagProvider.tsx

```diff
@@ -76,29 +76,29 @@ export const FeatureFlagsProvider = ({ children }: FeatureFlagsProviderProps): J
                 )
             }
 
-            const [major, minor, patch, rc] = targetVersion.split(/[.-]/)
-            const [targetMajor, targetMinor, targetPatch, targetRc] = nodeVersion.split(/[.-]/)
+            const [coreTargetVersion, targetVariant] = targetVersion?.split('-')
+            const [coreNodeVersion, nodeVariant] = nodeVersion?.split('-')
 
-            if (major !== targetMajor || minor !== targetMinor || patch !== targetPatch) {
-                return false
-            }
+            const [targetMajor, targetMinor, targetPatch] = coreTargetVersion.split('.').map(Number)
+            const [nodeMajor, nodeMinor, nodePatch] = coreNodeVersion.split('.').map(Number)
 
-            if (!rc) {
-                return true
+            if (targetMajor !== nodeMajor) {
+                return targetMajor < nodeMajor
             }
 
-            if (!targetRc) {
-                return false
+            if (targetMinor !== nodeMinor) {
+                return targetMinor < nodeMinor
             }
 
-            const rcNumber = parseInt(rc.replace('rc', ''))
-            const targetRcNumber = parseInt(targetRc.replace('rc', ''))
+            if (targetPatch !== nodePatch) {
+                return targetPatch < nodePatch
+            }
 
-            if (isNaN(rcNumber) || isNaN(targetRcNumber)) {
-                return false
+            if (nodeVariant) {
+                return !targetVariant || targetVariant <= nodeVariant
             }
 
-            return rcNumber >= targetRcNumber
+            return !targetVariant
```

Comment:
if nodeversion is 0.4.10 and targetversion is 0.4.10-rc1 , it will return false. I think it should be return true, as if we reach this point then we have major, minor and patch equal